reporter/attrmgr.go

```diff
@@ -0,0 +1,62 @@
+package reporter
+
+import (
+	"fmt"
+
+	common "go.opentelemetry.io/proto/otlp/common/v1"
+	profiles "go.opentelemetry.io/proto/otlp/profiles/v1experimental"
+)
+
+// AttrIndex is an index in the `Profile.attribute_table`.
+type AttrIndex = uint64
+
+// AttrTableManager maintains the `Profile.attribute_table` field for the
+// `profile.Profile` that it was created for. Attributes are automatically
+// deduplicated by the manager.
+type AttrTableManager struct {
+	// indices maps compound keys to the indices in the attribute table.
+	indices map[string]AttrIndex
+
+	// profile being built right now.
+	profile *profiles.Profile
+}
+
+func NewAttrTableManager(profile *profiles.Profile) *AttrTableManager {
```

Comment:
`*profiles.Profile` is internal to the package reporter. How are users of the package supposed to create it?
How should the integration look like, for a different kind of reporter, like for the OTel collector?  see https://github.com/open-telemetry/opentelemetry-ebpf-profiler/pull/208

From the updated testcase (`TestGetSampleAttributes`), where this API is used, it looks more like this API build a parallel `*profiles.Profile` on top of an existing one.

Reply:
How is this private? It's a public type from the OTEL proto lib?

> From the updated testcase (TestGetSampleAttributes), where this API is used, it looks more like this API builds a parallel *profiles.Profile on top of an existing one just with attributes but without traces.

No, it's just populating the attribute table portion. I'll apply the change that Tim suggested [here](https://github.com/open-telemetry/opentelemetry-ebpf-profiler/pull/237#discussion_r1839868496) to make this clearer.